arcgis-android-toolkit/src/main/java/com/esri/arcgisruntime/toolkit/floorfilter/FloorFilterManager.kt

```
@@ -0,0 +1,362 @@
+                val temp = _selectedLevelId
+                _selectedLevelId = null
+                selectedLevelId = temp
+                filterMap()
+
+                setupDone.invoke()
+            }
+        }
+        floorManager?.addDoneLoadingListener(doneLoadingListener)
+        floorManager?.loadAsync()
+    }
+
+    /**
+     * Removes the attached [GeoView].
+     *
+     * @since 100.13.0
+     */
+    fun clearGeoView() {
+        clearMapFilter()
+        this.geoView = null
+        this.floorManager = null
+    }
+
+    /**
+     * Zooms to the selected [FloorFacility]. If no [FloorFacility] is selected, it will zoom to
+     * the selected [FloorSite].
+     *
+     * @since 100.13.0
+     */
+    internal fun zoomToSelection() {
+        if (!selectedFacilityId.isNullOrBlank()) {
+            zoomToFacility()
+        } else if (!selectedSiteId.isNullOrBlank()) {
+            zoomToSite()
+        }
+    }
+
+    /**
+     * Zooms to the selected [FloorSite].
+     *
+     * @since 100.13.0
+     */
+    private fun zoomToSite() {
+        zoomToExtent(geoView, getSelectedSite()?.geometry?.extent)
+    }
+
+    /**
+     * Zooms to the selected [FloorFacility].
+     *
+     * @since 100.13.0
+     */
+    private fun zoomToFacility() {
+        zoomToExtent(geoView, getSelectedFacility()?.geometry?.extent)
+    }
+
+    /**
+     * Zooms the [geoView] to the [envelope].
+     *
+     * @since 100.13.0
+     */
+    private fun zoomToExtent(geoView: GeoView?, envelope: Envelope?, bufferFactor: Double = 1.25) {
+        if (geoView != null && envelope != null && !envelope.isEmpty) {
+            try {
+                val envelopeWithBuffer = Envelope(envelope.center, envelope.width * bufferFactor,envelope.height * bufferFactor)
+                if (!envelopeWithBuffer.isEmpty) {
+                    geoView.setViewpointAsync(Viewpoint(envelopeWithBuffer), 0.5f)
+                }
+            } catch (e: Exception) {
```

We should try to avoid "swallowing" exceptions. Do you anticipate any exceptions to be thrown? If yes, can we deal with those specific cases? If no, I suggest to remove the try/catch.

---

I do not anticipate any exceptions to be thrown. I will remove the try/catch.